docs/ABI/CallConvSummary.rst

```diff
@@ -0,0 +1,194 @@
++----------+---------+ by dyld, can be used    |          |          |          |
+| ``x17``  | ``ip1`` | freely otherwise)       |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x18``  |         | RESERVED **DO NOT USE** |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x19``  |         | Callee-saved register   |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x20``  |         | Callee-saved register   |          |          | ``self`` |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x21``  |         | Callee-saved register   |          |          | Error    |
+|          |         |                         |          |          | return   |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x22``- |         | Callee-saved registers  |          |          |          |
+| ``x28``  |         |                         |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x29``  | ``fp``  | Frame pointer           |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``x30``  | ``lr``  | Link register           |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``sp``   |         | Stack pointer           |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``v0``-  |         | Floating point/SIMD     |          |          |          |
+| ``v7``   |         | arguments 1-8           |          |          |          |
+|          |         | (also for return)       |          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+| ``v8``-  |         | Callee-saved registers  |          |          |          |
+| ``v15``  |         | (**lower 64-bits only**)|          |          |          |
++----------+---------+-------------------------+----------+----------+----------+
+
+Stack frame
+^^^^^^^^^^^
+
+The stack pointer is **16-byte aligned**.
+
++--------------+---------------+------------------------+
+|              | ``fp+8n+16``  | last memory argument   |
+|              |               |                        |
+|              | ...           | ...                    |
+|              |               |                        |
+|              | ``fp+16``     | memory argument 0 [1]_ |
++--------------+------------+--+------------------------+
+| ↓ Current Frame           |          ↑ Previous Frame |
++--------------+------------+--+------------------------+
+| entry ``sp`` | ``fp+8``      | saved ``lr``           |
```

Comment:
have you considered documenting the "entry `sp`" in the table's header? Does it simplify the table to get rid of this one almost empty column?

Reply:
It might simplify the table, but that would omit some useful information (namely where the stack pointer points on function entry). The position at which the stack pointer points is useful when you're looking at a leaf function, which might not set up its own stack frame. It's also a notable difference between ARM64 and x86, because of course ARM uses a link register and doesn't automatically push the return address. I'll add an explanatory note or two.